### PR TITLE
Kamu UI 377 change and addition of the variables and secrets tab

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**
Closes: https://github.com/kamu-data/kamu-web-ui/issues/377

Result:
image

image

### CODE REVIEW

File: src/app/dataset-view/additional-components/dataset-settings-component/dataset-settings.component.ts [hunk 1]
```diff
@@ -105,7 +105,10 @@ export class DatasetSettingsComponent extends BaseComponent implements OnInit {
             case SettingsTabsEnum.COMPACTION:
                 return this.datasetBasics.kind === DatasetKind.Root;
             case SettingsTabsEnum.VARIABLES_AND_SECRETS:
-                return this.appConfigService.featureFlags.enableDatasetEnvVarsManagment;
+                return (
+                    this.appConfigService.featureFlags.enableDatasetEnvVarsManagment &&
```
Comment (sergiimk) [inline src/app/dataset-view/additional-components/dataset-settings-component/dataset-settings.component.ts:109]: It's really nice that you have a flag to disable the feature entirely - we should use this pattern more!

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html [hunk 2]
```diff
@@ -47,12 +56,12 @@
...
             <div class="input-group">
                 <input
-                    [type]="row?.isSecret && !isShowExposedValue ? 'password' : 'text'"
+                    [type]="isSecretControl.value && !isShowExposedValue ? 'password' : 'text'"
                     class="form-control"
                     placeholder="Please enter value..."
                     formControlName="value"
                     data-test-id="value"
-                    [readonly]="row?.isSecret && !isShowExposedValue ? true : false"
+                    [readonly]="row && row.isSecret ? true : false"
```
Comment (sergiimk) [inline src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html:64]: Not sure I understand this.

If `row` is not null this means we're in the "edit" mode, right?

Why are we making value read-only when editing secrets?

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/dataset-settings-secrets-manager-tab.component.html [hunk 3]
```diff
@@ -1,12 +1,19 @@
 <div class="w-75 mb-100">
     <h2>Variables and secrets</h2>
     <mat-divider class="mt-1"></mat-divider>
-    <p class="text-muted mt-2 mb-2">Key-value pairs associated with dataset</p>
+    <p class="text-muted mt-2 mb-2">
+        Variables and secrets allow you to provide configuration parameters to ingest flows. Variables can be used for
+        non-sensitive values like page limits while secrets can be used for sensitive values like API keys and auth
+        tokens. Learn more about ingest parametrization
+        <a target="_blank" href="https://docs.kamu.dev/cli/ingest/input-formats/?highlight=env#dealing-with-api-keys"
+            >here </a
```
Comment (sergiimk) [inline src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/dataset-settings-secrets-manager-tab.component.html:9]: would this underline a blank space after "here"?

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html (hunk 2 above)
Comment (sergiimk) [reply src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html:64]: My understanding was that secrets are extractable (can be viewed after creation) and editable. Correct me if I'm wrong @rmn-boiko

Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html:64]: Yes, I forgot that we can edit secret values. Then I delete the attribute **readonly**

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/dataset-settings-secrets-manager-tab.component.html (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/dataset-settings-secrets-manager-tab.component.html:9]: No, it won't. The space is added by the linter.

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**